modules/bibauthorid/lib/bibauthorid_hoover.py

```diff
@@ -0,0 +1,746 @@
+
+from invenio.dbquery import run_sql
+from invenio.bibauthorid_dbinterface import get_existing_authors, \
+    get_canonical_name_of_author, get_papers_of_author, \
+    get_all_paper_data_of_author, \
+    get_signatures_of_paper, get_author_info_of_confirmed_paper, \
+    _get_external_ids_from_papers_of_author, get_claimed_papers_of_author, \
+    get_inspire_id_of_signature, populate_partial_marc_caches, move_signature, \
+    add_external_id_to_author, get_free_author_id, get_inspire_id_of_author, \
+    get_orcid_id_of_author, destroy_partial_marc_caches, get_name_by_bibref
+from invenio.bibauthorid_general_utils import memoized
+import invenio.bibauthorid_dbinterface as db
+import invenio.bibauthorid_config as bconfig
+from invenio.bibauthorid_webapi import get_hepnames, add_cname_to_hepname_record
+from invenio.bibcatalog import BIBCATALOG_SYSTEM
+from invenio.bibtask import write_message, task_sleep_now_if_required
+from invenio.bibauthorid_hoover_exceptions import *
+
+
+try:
+    from collections import defaultdict
+except ImportError:
+    from invenio.bibauthorid_general_utils import defaultdict
+
+class HooverStats(object):
+    tickets_raised = 0
+    new_ids_found = 0
+    connections_to_hepnames = 0
+    move_signature_calls = 0
+
+    @classmethod
+    def report_results(cls):
+        log = ("Tickets raised: {0} "
+               "New identifiers found: {1} "
+               "Connections to hepnames performed: {2} "
+               "Signatures attempted to move: {3}").format(cls.tickets_raised,
+                       cls.new_ids_found, cls.connections_to_hepnames, cls.move_signature_calls)
+        write_message(log, verbose=4)
+        return (cls.tickets_raised, cls.new_ids_found, cls.connections_to_hepnames, cls.move_signature_calls)
+
+def open_rt_ticket(e, debug_log=False, queue='Test'):
+    """Take an exception e and, if allowed by the configuration,
+    open a ticket for that exception.
+
+    Arguments:
+    e -- the exception to be reported
+    """
+    global ticket_hashes
+    ticket_hash = e.hash()
+    subject = e.get_message_subject() + ' ' + ticket_hash
+    body = e.get_message_body()
+    if debug_log:
+        debug = e.__repr__() + '\n' + \
+            '\n'.join([
+                str(key) + " " +
+                str(value) for key, value in vars(e).iteritems()])
+    else:
+        debug = ''
+    if rt_ticket_report:
+        if ticket_hash not in ticket_hashes.iterkeys():
+            ticket_id = BIBCATALOG_SYSTEM.ticket_submit(uid=None,
+                                                        subject=subject,
+                                                        recordid=e.recid,
+                                                        text=body +
+                                                        '\n Debugging\
+                                                        information: \n' +
+                                                        debug,
```

Comment:
Better to add the prefix `'\n Debugging information: \n'` the debug variable, so that if `debug_log` is `False`, there is no useless message.